lib/spack/spack/build_environment.py

```diff
@@ -404,12 +404,15 @@ def set_build_environment_variables(pkg, env, dirty):
 
 def _set_variables_for_single_module(pkg, module):
     """Helper function to set module variables for single module."""
-    # number of jobs spack will build with.
-    jobs = spack.config.get('config:build_jobs') or multiprocessing.cpu_count()
+
+    # The package required a serial build by setting parallel=False
     if not pkg.parallel:
-        jobs = 1
-    elif pkg.make_jobs:
-        jobs = pkg.make_jobs
+        spack.config.set('config:build_jobs', 1, scope='_builtin')
```

Comment:
There should be no `config.set` here, as it can affect other builds.  This should only affect `jobs` for the current package build.

You don't actually even need to set this here, as defaults for `_builtin` should be taken care of here:

https://github.com/spack/spack/blob/develop/lib/spack/spack/config.py#L103

In general, do *not* spread the responsibility for setting defaults to all parts of the code that access the config setting -- keep that in one place (`config_defaults` in `config.py`).

Reply:
Right. I forgot here that the config object is persistent. It's also kind of bad that this wasn't caught by unit tests, so I should probably add one for the case of a single package override within a DAG.

Reply:
Unit test added in cad37ea